pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/LoadBalancerTest.java

```diff
@@ -321,8 +322,12 @@ public void testBrokerRanking() throws Exception {
         }
 
         for (int i = 0; i < BROKER_COUNT; i++) {
-            Method updateRanking = Whitebox.getMethod(SimpleLoadManagerImpl.class, "updateRanking");
-            updateRanking.invoke(pulsarServices[i].getLoadManager().get());
+            Method method = Whitebox.getMethod(SimpleLoadManagerImpl.class, "getUpdateRankingHandle");
+            final int index = i;
+            Awaitility.await().until(() -> {
+                Object invoke = method.invoke(pulsarServices[index].getLoadManager().get());
```

Comment:
you may capture` pulsarServices[index] `out of the loop, instead of capturing `i`

Reply:
yes, right.